drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -381,6 +387,22 @@ static int stm32_clock_control_init(struct device *dev)
 
 #elif CONFIG_CLOCK_STM32_SYSCLK_SRC_HSE
 
+	old_hclk_freq = HAL_RCC_GetHCLKFreq();
+
+	/* Calculate new SystemCoreClock variable based on HSE freq */
+	new_hclk_freq = __LL_RCC_CALC_HCLK_FREQ(CONFIG_CLOCK_STM32_HSE_CLOCK,
+						hclk_prescaler);
+#if defined(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC)
+	__ASSERT(new_hclk_freq == CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
+			 "Config mismatch HCLK frequency %u %u",
+			 CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC, new_hclk_freq);
+#endif
+
+	/* If freq increases, set flash latency before any clock setting */
+	if (new_hclk_freq > old_hclk_freq) {
+		LL_SetFlashLatency(new_hclk_freq);
```

Comment:
Unfortunately, today this function is only available for a small set of series (L4, L5, G4).
For other series it is available under its private version: UTILS_SetFlashLatency. They will be progressively converted to the public version, but this is not done yet. 
What you can do is to update hal stm32 module to get all the series converted to the new public API.

Reply:
Thanks, @erwango 
Indeed I realized that the SetFlashLatency function was available only for L4/L5/G4, and I also had a look to the RM of the WB55 series, for which I got the fails, and the flash latencies management looked so close to the L4 series we're using... Even setting up a project for the WB55 using Stm32CubeIde confirmed that the interfaces were very close.
But your hint is now welcome, I hadn't realized yet that the other series had the same function as private. Tomorrow I'll work at this :-)